scipy/optimize/tests/test_minpack.py

```diff
@@ -394,6 +394,13 @@ def test_concurrent_with_gradient(self):
 
 
 class TestCurveFit(object):
+    def test_empty_array_with_bounds(self):
+        # This test is regarding the issue #9864
+        # Test that Curve_fit checks for empty input data
+        # if called with bounds
+        # Calling this function will produce a meaningful error message.
+        assert_raises(TypeError, curve_fit, lambda x, a: a*x, [], [], bounds=(1, 2))
```

Comment:
this will fail, you want `ValueError`

Reply:
Yea, I will change it. What about other changes? Any more changes in `check_len()`?